modules/social_features/social_group/modules/social_group_request/src/Form/GroupRequestMembershipRequestForm.php

```diff
@@ -99,56 +55,66 @@ public function getFormId() {
-    ];
+    /** @var \Drupal\group\Entity\Storage\GroupRelationshipTypeStorageInterface $storage */
+    $storage = $this->entityTypeManager->getStorage('group_content_type');
+    $group_type_id = (string) $group->getGroupType()->id();
+    $relation_type_id = $storage->getRelationshipTypeId($group_type_id, 'group_membership_request');
+    $group_content = $this->entityTypeManager->getStorage('group_content')->create([
+      'type' => $relation_type_id,
+      'gid' => $group->id(),
+      'entity_id' => $this->currentUser()->id(),
+      'grequest_status' => GroupMembershipRequest::REQUEST_PENDING,
+    ]);
+    $this->setEntity($group_content);
 
-    $form['message'] = [
-      '#type' => 'textarea',
-      '#title' => $this->t('Message'),
-    ];
+    $form = parent::buildForm($form, $form_state);
 
-    $form['actions']['submit'] = [
-      '#type' => 'submit',
-      '#value' => $this->t('Send request'),
-      '#button_type' => 'primary',
-    ];
+    if (isset($form['field_grequest_message'])) {
+      $form['description'] = [
+        '#type' => 'html_tag',
+        '#tag' => 'p',
+        '#value' => $this->t('You can leave a message in your request. Only when your request is approved, you will receive a notification via email and notification center.'),
+        '#weight' => -10,
+      ];
+    }
 
     return $form;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function submitForm(array &$form, FormStateInterface $form_state) {
-    /** @var \Drupal\group\Entity\Storage\GroupRelationshipTypeStorageInterface $storage */
-    $storage = $this->entityTypeManager->getStorage('group_content_type');
-    $group_type_id = (string) $this->group->getGroupType()->id();
-    $relation_type_id = $storage->getRelationshipTypeId($group_type_id, 'group_membership_request');
+  protected function actions(array $form, FormStateInterface $form_state) {
+    $actions = parent::actions($form, $form_state);
+    $actions['submit']['#value'] = $this->t('Send request');
+    return $actions;
+  }
 
-    $group_content = $this->entityTypeManager->getStorage('group_content')->create([
-      'type' => $relation_type_id,
-      'gid' => $this->group->id(),
-      'entity_id' => $this->currentUser()->id(),
-      'grequest_status' => GroupMembershipRequest::REQUEST_PENDING,
-      'field_grequest_message' => $form_state->getValue('message'),
-    ]);
-    $result = $group_content->save();
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    parent::submitForm($form, $form_state);
 
-    if ($result) {
-      $this->messenger()->addMessage($this->t('Your request has been sent successfully'));
-    }
-    else {
-      $this->messenger()->addError($this->t('Error when creating a request to join'));
-    }
+    /** @var \Drupal\group\Entity\GroupRelationshipInterface $entity */
+    $entity = $this->entity;
```

Comment:
Prefer using assert here. See [development guideline 2.2 assert over `@var`](https://open-social.slite.com/app/docs/dm052z21PPpHST#02da03b7) for details.